lib/dgram.js

```diff
@@ -97,19 +98,27 @@ function newHandle(type, lookup) {
   throw new ERR_SOCKET_BAD_TYPE();
 }
 
+// Follow the net.js module and ignore fd if it's not a valid number.
+function isPossibleValidFD(fd) {
+  return typeof fd === 'number' && fd > 0;
```

Comment:
`&& isFinite(fd)` ?

Reply:
Or follow the [net.js](https://github.com/nodejs/node/blob/master/lib/net.js#L96) and use `validateInt32`. I will change it.